src/query.rs

```diff
@@ -1655,6 +1655,45 @@ fn assert_distinct<const N: usize>(entities: &[Entity; N]) {
         }
     }
 }
+
+/// A borrow of a [`World`](crate::World) sufficient to execute the query `Q`, and immediately exposes a `View`
+/// of it.
+///
+/// Note that borrows are not released until this object is dropped.
+///
+/// This struct is a thin wrapper around [`View`](crate::View). See it for more documentation.
+pub struct ViewBorrowed<'w, Q: Query> {
+    _query: QueryBorrow<'w, Q>,
+    view: View<'w, Q>,
+}
+
+impl<'w, Q: Query> ViewBorrowed<'w, Q> {
+    pub(crate) fn new(world: &'w World) -> Self {
+        let mut query = world.query::<Q>();
+        query.borrow();
+        let view = unsafe { View::<Q>::new(world.entities_meta(), world.archetypes_inner()) };
+
+        Self {
+            _query: query,
+            view,
+        }
+    }
+}
+
+impl<'w, Q: Query> core::ops::Deref for ViewBorrowed<'w, Q> {
+    type Target = View<'w, Q>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.view
+    }
+}
+
+impl<'w, Q: Query> core::ops::DerefMut for ViewBorrowed<'w, Q> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
```

Comment:
I think this may be unsound after all. A caller could create a `View<'w, Q>` from an unrelated world, then `mem::swap` it with this one, then drop the `ViewBorrow`, leading to dangling references, violating hecs' preconditions. That procedure could then be repeated to create illegally aliasing references, creating immediate UB.

The only way I see to avoid this is to ensure that `View` itself borrows from the `ViewBorrow`, not the `World`. I don't think that's expressible through `DerefMut`, and with an explicit getter we're back at exactly the same ergonomics as creating a `query` first.

The only sound path forwards I see is to manually replicate the `View` API directly on `ViewBorrow`. We could get up to some trait shenanigans, but I'm not a fan of requiring users to `use` a trait to call routine methods on a concrete type. This is mildly annoying to do, but IMO library boilerplate is a low cost to pay for a better user experience, and it's a tradeoff hecs already makes in various other places (e.g. prepared queries).